pkg/flags/ipaddressport.go

```diff
@@ -32,22 +32,26 @@ type IPAddressPort struct {
 func (a *IPAddressPort) Set(arg string) error {
 	arg = strings.TrimSpace(arg)
 
-	parts := strings.SplitN(arg, ":", 2)
-	if len(parts) != 2 {
-		return errors.New("bad format in address specification")
+	host, port, err := net.SplitHostPort(arg)
+	if err != nil {
+		return err
 	}
 
-	if net.ParseIP(parts[0]) == nil {
+	if net.ParseIP(host) == nil {
 		return errors.New("bad IP in address specification")
 	}
 
-	port, err := strconv.Atoi(parts[1])
+	port_number, err := strconv.Atoi(port)
 	if err != nil {
 		return errors.New("bad port in address specification")
 	}
 
-	a.IP = parts[0]
-	a.Port = port
+	if arg[0] == '[' {
+		a.IP = "[" + host + "]"
+	} else {
+		a.IP = host
+	}
+	a.Port = port_number
```

Comment:
I think `a.IP = host` is better. We should use `net.JoinHostPort` in `String` func to get the address.